aurora/aurora-war/src/main/java/org/silverpeas/looks/aurora/LookAuroraHelper.java

```diff
@@ -352,6 +351,10 @@ public NewsList getNews() {
 
   public NextEvents getNextEvents() {
     List<String> allowedComponentIds = asList(getAllowedComponentIds("home.events.appId"));
+    return getNextEvents(allowedComponentIds);
+  }
+
+  public NextEvents getNextEvents(List<String> allowedComponentIds) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 25 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS3776)